### PR TITLE
HDS-2069 first steps to new Select, Group component

### PR DESCRIPTION
**Pull request by NikoHelle**
## Description

**Target branch is the feature branch of new Select/MultiSelect**

The first step towards simpler code with complex components: `Group`-component and its `controller`.

The `Group` shares its data with its children. Data is stored to a `controller`. Controller also handles props of each  children in the `Group`. This way all changes happen in one place and all props are in same place and data is shared.

`Group` props include a `propSetter` and `onChange` functions. `PropSetter` returns props for each child component and `onChange` handles data changes.

React hooks and context is not used, so most of the code would work in vanilla js and web components.

This PR has only initial code and does not take into account all user scenarios or optimise anything - except DOM updates are tested well.

## Related Issue

[HDS-2069](https://helsinkisolutionoffice.atlassian.net/browse/HDS-2069)

Closes nothing yet.

## Motivation and Context

Created to make complex components simpler. PropSetter has helpers for UI events and later on also aria-attributes can be added with common helpers. All components can basically access all data in a very simple way.

## How Has This Been Tested?

Added unit tests and basic stories.

[Demo](https://city-of-helsinki.github.io/hds-demo/hds-2069-select/?path=/story/components-group--accordion)

## Add to changelog
No additions yet.

[HDS-2069]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-2069?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by laurakarhu** [comment]
What should I be looking at in the demo? There is no select available?

File: packages/react/src/components/group/Group.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+import { TextInput } from '../textInput/TextInput';
+import { Button } from '../button/Button';
+import { ChangeHandlerProps, Controller, GroupChild, GroupChildFunction, GroupProps } from './utils';
+import { Group, RenderWithController } from './Group';
+import { createOnClickListener, createInputOnChangeListener } from './utils/propSetterHelpers';
+import { IconArrowDown, IconArrowUp } from '../../icons';
+import { ForwardController } from './utils/forwardController';
+
+export default {
+  component: Group,
```
Comment (laurakarhu) [inline packages/react/src/components/group/Group.stories.tsx:12]: Some comments here what the group is for?

**Review by laurakarhu — COMMENTED** [review]
Some first remarks

File: packages/react/src/components/group/Group.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+  const JoinedValues: GroupChildFunction = ({ controller }: { controller: Controller }) => {
+    const allProps = Object.values(controller.getData() as Record<string, string>).join('+');
+    return <p>All data joined: {allProps}</p>;
+  };
+
+  return (
+    <Group {...props}>
+      <TextInput id="text" key="text" label="Input" />
+      <TextInput label="Reversed input:" disabled id="reverse" data-groupid="reverse" key="rev" />
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/group/Group.stories.tsx:47]: Thx, didn't know that! I'll add `-hds-`

**Comment by NikoHelle** [comment]
> What should I be looking at in the demo? There is no select available?

https://city-of-helsinki.github.io/hds-demo/hds-2069-select/?path=/story/components-group--accordion

There is just simple components: accordion and numeric stepper. And a text input demo how to share data.

Select is the next step(s)

**Comment by NikoHelle** [comment]
Now that main parts of the Select are done, this Group system does not seem be beneficial enough. It adds a lot of new code, but does not make coding so much easier that it should be kept.

**Pull request closed without merging**